src/native-state-drm.cpp

```diff
@@ -197,27 +199,7 @@ NativeStateDRM::init_gbm()
 bool
 NativeStateDRM::init()
 {
-    // TODO: Replace this with something that explicitly probes for the loaded
-    // driver (udev?).
-    static const char* drm_modules[] = {
-        "i915",
-        "nouveau",
-        "radeon",
-        "vmgfx",
-        "omapdrm",
-        "exynos"
-    };
-
-    unsigned int num_modules(sizeof(drm_modules)/sizeof(drm_modules[0]));
-    for (unsigned int m = 0; m < num_modules; m++) {
-        fd_ = drmOpen(drm_modules[m], 0);
-        if (fd_ < 0) {
-            Log::debug("Failed to open DRM module '%s'\n", drm_modules[m]);
-            continue;
-        }
-        Log::debug("Opened DRM module '%s'\n", drm_modules[m]);
-        break;
-    }
+    fd_ = open("/dev/dri/card0", O_RDWR);
```

Comment:
If the `vgem` driver does not expose a `/dev/dri/card0` device node, then either the node is already defined by another driver, in which the `vgem` node will be `/dev/dri/card1`, or the driver should be fixed, IMHO.

The whitelist system should be avoided at all costs, as you can't maintain a hardcoded list of past, present, future and exotic DRM modules.

If the DRM driver does not expose `/dev/dri/card0` anymore, then either :
* the sysadmin used udev to rename the dev entry;
* another DRM driver has already created this device node, in which case `/dev/dri/card1` should be available;
* the device driver has serious issues;
* the standard completely changed.

I don't know where this whitelist system comes from. KMS/DRM tutorials clearly define that you should just open `/dev/dri/card0` : [Writing standalone programs with EGL and KMS (updated)](http://virtuousgeek.org/blog/index.php/jbarnes/2011/10/31/writing_stanalone_programs_with_egl_and_)

This is also how the MESA guys do it now : https://github.com/robclark/kmscube/blob/master/kmscube.c#L76

That said, I can :
* split the patch in two,
* use this whitelist system as a fallback mechanism, for old device drivers that do not expose /dev/dri/card0,
* provide a way to define the device node to use for DRM drivers, for multiple graphic cards systems, like this here https://github.com/dvdhrm/docs/blob/master/drm-howto/modeset.c